File: nutrinova-api/NutrinovaData/Features/Reports/NutrientGoalReportItem.cs
namespace NutrinovaData.Features.Reports;

public class NutrientGoalReportItem
{
  public int NutrientId { get; set; }

  public string NutrientName { get; set; } = string.Empty;

  public UnitOption PreferredUnit { get; set; } = new UnitOption();

  public GoalTargetAmount CustomTargetAmount { get; set; } = default!;

  public GoalTargetAmount RecommendedTargetAmount { get; set; } = default!;

  public decimal ConsumedAmount { get; set; }

  public NutrientGoalStatus GoalStatus { get; set; } = default;
}

public enum NutrientGoalStatus
{
  NotStarted,
  NotMet,
  Met,
  Exceeded,
}

File: nutrinova-ui/src/app/(authorized)/goals/_models/NutrientGoalReportItem.ts
import { UnitOption } from "../../food/_models/unitOption";

export interface NutrientGoalReportItem {
  nutrientId: number;
  nutrientName: string;
  preferredUnit: UnitOption;
  customTargetAmount: GoalTargetAmount;
  recommendedTargetAmount: GoalTargetAmount;
  consumedAmount: number;
  goalStatus: NutrientGoalStatus;
}

export interface DailyNutrientGoalReport {
  date: Date;
  nutrientGoalReportItems: NutrientGoalReportItem[];
}

export interface GoalTargetAmount {
  lowerLimit: number;
  upperLimit: number;
  maxLimit: number;
  type: string;
}

export enum NutrientGoalStatus {
  NotStarted = 0,
  NotMet = 1,
  Met = 2,
  Exceeded = 3,
}